windows/vtpm_cold_add_remove/cold_add_vtpm.yml

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+- name: "Handle VM without vTPM device situation"
+  when: not vm_with_vtpm_before
+  block:
+    - name: "Power off instant cloned VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'powered-off'
+    - name: "Add key provider on vCenter server"
+      include_tasks: ../../common/vcenter_add_key_provider.yml
+      vars:
+        vc_cert_path: "{{ current_test_log_folder }}"
+      when:
+        - key_provider_type is defined and key_provider_type
+        - not (key_provider_added is defined and key_provider_added)
+
+    - name: "Add vTPM device to cloned VM"
+      include_tasks: ../../common/vm_add_remove_vtpm.yml
+      vars:
+        vtpm_operation: 'present'
+    - name: "Pause 10 seconds after adding vTPM device"
+      ansible.builtin.pause:
+        seconds: 10
+
+- name: "Power on cloned VM"
+  include_tasks: ../../common/vm_set_power_state.yml
+  vars:
+    vm_power_state_set: 'powered-on'
+
+- name: "Get cloned VM MAC address"
+  include_tasks: ../../common/vm_wait_primary_nic_mac.yml
+- name: "Get cloned VM IP address"
+  include_tasks: ../../common/vm_get_ip.yml
+  vars:
+    vm_get_ip_timeout: 600
+- name: "Print VM IP address"
+  ansible.builtin.debug:
+    msg:
+      - "Cloned VM IP address: {{ vm_guest_ip }}"
+      - "Parent VM IP address: {{ vm_ip_clone_from }}"
```

Comment:
It's better to check the cloned VM's IP and parent VM's IP are different here or the cloned and parent VM's primary mac address here. Because the cloned VM would has same mac address with parent VM, and then VC would take few seconds or more time to resolve MAC address conflicting. If it happens that MAC/IP addresses on cloned VM and parent VM are same at here, the following tasks might be performed on wrong VM. 
This happens at Linux test case `pvrdma_network_device_ops.yml`. So I suggest you considering the worst case here and avoid such kind of failure due to MAC/IP address conflicts.

Reply:
Ok, will update here to avoid this situation. Thanks.